src/new/sendTransaction.ts

```diff
@@ -494,4 +509,162 @@ class SendTransaction extends EventEmitter {
   }
 }
 
-export default SendTransaction;
+/**
+ * Check the tx data and propose inputs and outputs to complete the transaction.
+ * We will only check a single token
+ *
+ * @param {IStorage} storage
+ * @param {IDataTx} dataTx
+ * @param {IUtxoSelectionOptions} options
+ */
+export async function prepareSendTokensData(
+  storage: IStorage,
+  dataTx: IDataTx,
+  options: IUtxoSelectionOptions = {},
+): Promise<Pick<IDataTx, 'inputs'|'outputs'>> {
+  const token = options.token || HATHOR_TOKEN_CONFIG.uid;
+  const utxoSelection = options.utxoSelectionMethod || bestUtxoSelection;
+  const newtxData: Pick<IDataTx, 'inputs'|'outputs'> = { inputs: [], outputs: [] };
+  let outputAmount = 0;
+
+  // Calculate balance for the token on the transaction
+  for (const output of dataTx.outputs) {
+    if (isDataOutputCreateToken(output)) {
+      // This is a mint output
```

Comment:
This is only because the `output.token` property does not exist on `type: 'mint'` but it should also be ignored if the output was a `type: 'melt'` which would make this an HTR withdrawal, so no input would be needed for the withdrawal.

This check is mostly for the type since the facade does not allow for token operations or authority inputs/outputs of any kind when using the inputs and outputs format, so we will not have a mint or melt operation here in any way.